### PR TITLE
Add support for branch run type in EAMxx

### PR DESCRIPTION
**Pull request by bartgol**
It did not require much work, basically just grab the restart case name from mct rather than XML file. I piggy backed a small cleanup of the trace gases workaround struct.

### CODE REVIEW

File: components/eamxx/src/control/atmosphere_driver.cpp [hunk 1]
```diff
@@ -243,7 +257,7 @@ void AtmosphereDriver::create_grids()
   // To avoid having to pass the same data twice in the input file,
   // we have the AD add the IC file name to the GM params
   const auto& ic_pl = m_atm_params.sublist("initial_conditions");
-  if (m_case_t0<m_run_t0) {
+  if (m_run_type==RunType::Restart) {
```
Comment (bartgol) [inline components/eamxx/src/control/atmosphere_driver.cpp:260]: These checks are equivalent, but I thought using the enum may be more robust if we later decide to distinguish Continue from Branch, or maybe support Hybrid (which I still don't understand).

File: components/eamxx/src/mct_coupling/atm_comp_mct.F90 [hunk 2]
```diff
@@ -149,7 +150,8 @@ subroutine atm_init_mct( EClock, cdata, x2a, a2x, NLFilename )
          dom=dom_atm, &
          infodata=infodata)
     call seq_infodata_getData(infodata, atm_phase=phase, start_type=run_type, &
-                              username=username, case_name=caseid, hostname=hostname)
+                              username=username, case_name=caseid, &
+                              rest_case_name=rest_caseid, hostname=hostname)
```
Comment (bartgol) [inline components/eamxx/src/mct_coupling/atm_comp_mct.F90:154]: @rljacob Is `rest_case_name` always defined, even if this is a "regular" restart (not branch)? I assumed yes. That is, for a simple CONTINUE_RUN, I expect rest_case_name==case_name...

File: components/eamxx/src/control/atmosphere_driver.cpp [hunk 3]
```diff
@@ -153,17 +153,31 @@ init_scorpio(const int atm_id)
...
+
   // Initialize time stamps
   m_run_t0 = m_current_ts = run_t0;
   m_case_t0 = case_t0;
-}
-
 
+  switch (run_type) {
+    case 0:
+      m_run_type = RunType::Initial; break;
```
Comment (tcclevenger) [inline components/eamxx/src/control/atmosphere_driver.cpp:172]: Does this case need to require `case_t0 == run_t0`? Similar for restart, require `case_t0 < run_t0`?

Comment (bartgol) [reply components/eamxx/src/control/atmosphere_driver.cpp:172]: I suppose we could check. But I wonder if it matters at this point. The big need for case-vs-run t0 was to detect restarts. Now case_t0 is basically just used only as a "reference" time for output's time coord.

Comment (bartgol) [reply components/eamxx/src/control/atmosphere_driver.cpp:172]: Ah, wait. The output mgr still decides whether this is a rest run based on case_t0<run_t0. I'll pass run type to OM as well.

File: components/eamxx/src/mct_coupling/atm_comp_mct.F90 (hunk 2 above)
Comment (rljacob) [reply components/eamxx/src/mct_coupling/atm_comp_mct.F90:154]: Yes its read from the restart file.

File: components/eamxx/src/share/io/scream_output_manager.hpp [hunk 4]
```diff
@@ -79,12 +79,21 @@ class OutputManager
...
-                   const bool is_model_restart_output);
+                   const bool is_model_restart_output, const RunType run_type);
 
+  // This overloads are to make certain unit tests easier
+  void initialize (const ekat::Comm& io_comm, const ekat::ParameterList& params,
+                   const util::TimeStamp& run_t0, const util::TimeStamp& case_t0,
+                   const bool is_model_restart_output)
+  {
+    auto rt = case_t0<run_t0 ? RunType::Restart : RunType::Initial;
+    initialize(io_comm,params,run_t0,case_t0,is_model_restart_output,run_type);
```
Comment (tcclevenger) [inline components/eamxx/src/share/io/scream_output_manager.hpp:90]:
```suggestion
    initialize(io_comm,params,run_t0,case_t0,is_model_restart_output,rt);
```
Is this a typo?

**Review by tcclevenger — DISMISSED** [review]

**Comment by bartgol** [comment]
@brhillman confirmed that this works as expected. Letting the AT do its course and then merging.

**Review by tcclevenger — APPROVED** [review]

**Pull request merged**